### PR TITLE
Implement commute()

### PR DESCRIPTION
**Pull request by yamaguchi1024**

### CODE REVIEW

File: src/exo/API_scheduling.py [hunk 1]
```diff
@@ -715,6 +715,39 @@ def reorder_stmts(proc, block_cursor):
     loopir  = Schedules.DoReorderStmt(loopir, s1, s2).result()
     return Procedure(loopir, _provenance_eq_Procedure=proc)
 
+@sched_op([ExprCursorA(many=True)])
```
Comment (gilbo) [inline src/exo/API_scheduling.py:718]: Should this operation apply to a single location or many locations?

**Comment by gilbo** [comment]
This scheduling op should probably have a more specific name than `commute`.  Maybe `commute_expr` is sufficient.  But are `+` and `*` commutativity the only possible thing that the phrase `commute_expr` might mean?

File: src/exo/API_scheduling.py (hunk 1 above)
Comment (yamaguchi1024) [reply src/exo/API_scheduling.py:718]: I'm fine with both, do you have a suggestion?

**Comment by yamaguchi1024** [comment]
I'm fine with both `commute` and `commute_expr`, latter is a bit harder to type though

**Review by gilbo — APPROVED** [review]

**Pull request merged**